features/steps/steps/propertysets_qtys_units.py

```diff
@@ -375,3 +409,27 @@ def step_impl(context, inst, table, inst_type=None):
             # not a universal error. This is more IDS territory.
             # if not values:
             #     yield ValidationOutcome(inst=inst, expected= {"oneOf": accepted_data_type['instance']}, observed = {'value':None}, severity=OutcomeSeverity.ERROR)
+
+@gherkin_ifc.step(
+    "Its attribute .{attr_name}. must be defined [according to the table] 'valid_ConversionBasedUnits'"
+)
+def step_impl(context, inst, attr_name):
+    table = "valid_ConversionBasedUnits"
+    unit_definitions = get_table_definition(context.model.schema, table)
+    accepted_names = list(unit_definitions.keys())
+    match attr_name.upper():
+        case "NAME":
+            attr_value = getattr(inst, attr_name)
+            if attr_value not in accepted_names:
+                yield ValidationOutcome(inst=inst, expected=accepted_names, observed=attr_value, severity=OutcomeSeverity.ERROR)
+        case "CONVERSIONFACTOR":
+            unit_name = inst.Name
+            if unit_name in accepted_names:
+                inst_factor = inst.ConversionFactor.ValueComponent.wrappedValue
+                inst_si_unit = inst.ConversionFactor.UnitComponent
+                conv_unit_def = ConversionBasedUnitDefinition(**unit_definitions[unit_name])
+                expected_factor = ifcopenshell.util.unit.convert(value=conv_unit_def.ConversionFactor,from_unit=conv_unit_def.SIUnitName,from_prefix=conv_unit_def.SIUnitPrefix,to_unit=inst_si_unit.Name,to_prefix=inst_si_unit.Prefix)
+                if not math.isclose(a=inst_factor, b=expected_factor, rel_tol=1e-06, abs_tol=1e-06):
```

Comment:
I quickly checked just now:

> For the values to be considered close, the difference between them
> must be smaller than at least one of the tolerances.

Some values are awfully close to the abs_tol, so this is risky.

```
>>> y = 0.00001639
>>> x = 0.00001639 + (1.e-6-1.e-9)
>>> print(f"{x:.10f}")
0.0000173890
>>> print(f"{y:.10f}")
0.0000163900
>>> math.isclose(x, y, rel_tol=1e-06, abs_tol=1e-06)
True
```

Since these values can never be 0, I would say we only use a rel_tol.

```suggestion
                if not math.isclose(a=inst_factor, b=expected_factor, rel_tol=1e-06, abs_tol=0.):
```

(I didn't test this, but even if this is maybe not the correct solution, I think the current approach doesn't work well for small values and does need to change)